sympy/polys/tests/test_poly_hypothesis.py

```diff
@@ -0,0 +1,22 @@
+from hypothesis import given
+from hypothesis import strategies as st
+from sympy.abc import x
+from sympy.polys.polytools import Poly
+
+
+@given(
+    coefficients1=st.lists(st.integers()),
+    coefficients2=st.lists(st.integers()).filter(lambda x: not all(i == 0 for i in x)),
+)
+def test_poly_hypothesis(coefficients1, coefficients2):
+    # Integer case
+    f_z = Poly(coefficients1, x, domain="ZZ")
+    g_z = Poly(coefficients2, x, domain="ZZ")
+    remainder_z = f_z.rem(g_z)
+    assert g_z.degree() > remainder_z.degree() or remainder_z.degree() == 0
+
+    # Rational case
+    f_q = Poly(coefficients1, x, domain="QQ")
+    g_q = Poly(coefficients2, x, domain="QQ")
```

Comment:
I think that you should better figure out how to sample on rationals from 2-tuples of integers.

Reply:
The point here is just to make sure that both QQ and ZZ give valid answers, since they do different things even for pure integer polynomials. 

```
>>> gcd(Poly(4*x + 2, domain='ZZ'), Poly(2*x, domain='ZZ'))
Poly(2, x, domain='ZZ')
>>> gcd(Poly(4*x + 2, domain='QQ'), Poly(2*x, domain='QQ'))
Poly(1, x, domain='QQ')
```

(if the domain is a field, the result is monic, which could perhaps be tested here)